### PR TITLE
Add ability to generate OpenAPI spec from PureModelContext Rebased

### PR DESCRIPTION
**Pull request by janeenyamak1**
#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-openapi/legend-engine-xt-openapi-generation/pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
         <version>4.21.2-SNAPSHOT</version>
     </parent>
 
-    <artifactId>legend-engine-xt-openapi-generation</artifactId>
+    <artifactId>legend-engine-xt-openapi-generation-pure</artifactId>
```
Comment (hardikmaheshwari) [inline legend-engine-xts-openapi/legend-engine-xt-openapi-generation/pom.xml:26]: Is this expected?

**Review by hardikmaheshwari — CHANGES_REQUESTED** [review]

File: legend-engine-xts-openapi/legend-engine-xt-openapi-generation/pom.xml [hunk 2]
```diff
@@ -119,6 +202,16 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.core</groupId>
+            <artifactId>jersey-common</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-language-pure-dsl-service</artifactId>
```
Comment (hardikmaheshwari) [inline legend-engine-xts-openapi/legend-engine-xt-openapi-generation/pom.xml:213]: is it test scope, if not lets move above

File: legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiGenerationService.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+import static org.junit.Assert.assertNotNull;
+
+public class TestOpenApiGenerationService
+{
+    private ObjectMapper objectMapper = new ObjectMapperFactory().getNewStandardObjectMapperWithPureProtocolExtensionSupports();
```
Comment (hardikmaheshwari) [inline legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiGenerationService.java:35]: Lets use PureProtocolObjectMapperFactory

File: legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiGenerationService.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    @Test
+    public void shouldGenerateOpenApiSpecFromPureModelContext() throws IOException
+    {
+        String payload = IOUtils.toString(Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResource("pure_model_context_payload.json")), StandardCharsets.UTF_8);
+        assertNotNull(payload);
+        OpenApiGenerationInput input = objectMapper.readValue(payload, OpenApiGenerationInput.class);
+        ModelManager manager = new ModelManager(DeploymentMode.TEST);
+        OpenApiGenerationService service = new OpenApiGenerationService(manager);
+        Response response = service.generateOpenApi(input, null);
+        assertNotNull(response);
```
Comment (hardikmaheshwari) [inline legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiGenerationService.java:46]: we just checking something is generated and not the contents?

Comment (janeenyamak1) [reply legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiGenerationService.java:46]: now asserting the response entity to an expected result

Comment (hardikmaheshwari) [reply legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiGenerationService.java:48]: do we need assertNull. after checking equality with a string?

File: legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiArtifactGenerationExtension.java [hunk 5]
```diff
@@ -14,6 +14,7 @@
 
 package org.finos.legend.engine.generation;
 
+import org.finos.legend.engine.generation.extension.OpenApiArtifactGenerationExtension;
```
Comment (hardikmaheshwari) [inline legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiArtifactGenerationExtension.java:17]: ??

File: legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiGenerationService.java (hunk 4 above)
Comment (janeenyamak1) [reply legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiGenerationService.java:48]: will clean up the assert null

File: legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiArtifactGenerationExtension.java (hunk 5 above)
Comment (janeenyamak1) [reply legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiArtifactGenerationExtension.java:17]: yes, an instance of OpenApiArtifactGenerationExtension is being created in the test

**Pull request closed without merging**